platform-includes/migration/javascript-v8/important-changes/javascript.gatsby.mdx

```diff
@@ -6,6 +6,24 @@ If you need to support older versions of React, please use Sentry Gatsby SDK `7.
 
 <Include name="migration/javascript-v8/compatible-browsers" />
 
+### Updated error types to be `unknown` instead of `Error`.
+
+In v8, we are changing the `ErrorBoundary` error types returned from `onError`, `onReset`, `onUnmount`, and
+`beforeCapture`. to be `unknown` instead of `Error`. This more accurately matches behaviour of `componentDidCatch`, the
+lifecycle method the Sentry `ErrorBoundary` component uses.
+
+As per the [React docs on error boundaries](https://react.dev/reference/react/Component#componentdidcatch):
+
+> error: The `error` that was thrown. In practice, it will usually be an instance of `Error` but this is not guaranteed
+> because JavaScript allows to throw any value, including strings or even `null`.
+
+This means you will have to use `instanceof Error` or similar to explicitly make sure that the error thrown was an
+instance of `Error`.
+
+The Sentry SDK maintainers also went ahead and made a PR to update the
+[TypeScript definitions of `componentDidCatch`](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/69434) for the
+React package - this will be released with React 20.
```

Comment:
```suggestion
React package—this will be released with React 20.
```

Reply:
This was intentional usage of the dash as punctuation to form a pause, but it might not fit our style guide. Let's see what docs folks have to say.

Reply:
yea, I just replaced it with an emdash. We've been using them in a bunch of the js docs

Reply:
I feel like the em dash isn't necessary. You can just as easily start a new sentence.
```suggestion
React package. This will be released with React 20.
```